pycbc/live/snr_optimizer.py

```diff
@@ -0,0 +1,340 @@
+import argparse, numpy
+import time
+import logging
+import types
+from scipy.optimize import differential_evolution, shgo
+from pycbc import (
+    DYN_RANGE_FAC, waveform
+)
+from pycbc.types import zeros
+import pycbc.waveform.bank
+from pycbc.filter import matched_filter_core
+import pycbc.conversions as cv
+
+try:
+    import pyswarms as ps
+except:
+    ps = None
+
+# Set a minimum mass for points tried in optimization allowing for
+# minimal slop relative to the lightest template
+MIN_CPT_MASS = 0.99
+
+# Set a large maximum total mass
+MAX_MTOTAL = 500.
+
+Nfeval = 0
+start_time = time.time()
+
+def callback_func(Xi, convergence=0):
+    global Nfeval
+    logging.info("Currently at %d %s", Nfeval, convergence)
+    if (time.time() - start_time) > 360:
```

Comment:
Actually I'm not sure why this is here and why the specific value 360. Would be good to add a comment at least

Reply:
The optimization has an effective time-out of 6 minutes with this line, maybe that's a ridiculously long time but it helps to kill the optimizer if there is an issue causing hanging.